app/src/androidTest/java/org/mozilla/focus/telemetry/TelemetryWrapperTest.java

```diff
@@ -0,0 +1,571 @@
+
+import java.util.Locale;
+
+import static org.mozilla.focus.telemetry.TelemetryWrapper.Value.AUDIO;
+import static org.mozilla.focus.telemetry.TelemetryWrapper.Value.EME;
+import static org.mozilla.focus.telemetry.TelemetryWrapper.Value.MIDI;
+import static org.mozilla.focus.telemetry.TelemetryWrapper.Value.VIDEO;
+
+@RunWith(AndroidJUnit4.class)
+public class TelemetryWrapperTest {
+
+    @Rule
+    public ActivityTestRule<MainActivity> activityTestRule = new ActivityTestRule<>(MainActivity.class, true, false);
+
+    @Before
+    public void setUp() {
+        final Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        final String prefName = context.getString(R.string.pref_key_telemetry);
+        final SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
+        preferences.edit().putBoolean(prefName, false).apply();
+
+    }
+
+
+    @Test
+    public void toggleFirstRunPageEvent() {
+        TelemetryWrapper.toggleFirstRunPageEvent(false);
+    }
+
+    @Test
+    public void finishFirstRunEvent() {
+        TelemetryWrapper.finishFirstRunEvent(3000000);
+    }
+
+    @Test
+    public void browseIntentEvent() {
+        TelemetryWrapper.browseIntentEvent();
+    }
+
+    @Test
+    public void textSelectionIntentEvent() {
+        TelemetryWrapper.textSelectionIntentEvent();
+    }
+
+    @Test
+    public void launchByAppLauncherEvent() {
+        TelemetryWrapper.launchByAppLauncherEvent();
+    }
+
+    @Test
+    public void launchByHomeScreenShortcutEvent() {
+        TelemetryWrapper.launchByHomeScreenShortcutEvent();
+    }
+
+    @Test
+    public void launchByTextSelectionSearchEvent() {
+        TelemetryWrapper.launchByTextSelectionSearchEvent();
+    }
+
+    @Test
+    public void launchByExternalAppEvent() {
+        TelemetryWrapper.launchByExternalAppEvent();
+    }
+
+    @Test
+    public void settingsEvent() {
+        activityTestRule.launchActivity(new Intent());
+        SharedPreferences pm = PreferenceManager.getDefaultSharedPreferences(activityTestRule.getActivity());
+        for (String key : pm.getAll().keySet()) {
+            TelemetryWrapper.settingsEvent(key, Boolean.TRUE.toString());
+        }
+
+    }
+
+    @Test
+    public void settingsClickEvent() {
+        activityTestRule.launchActivity(new Intent());
+        SharedPreferences pm = PreferenceManager.getDefaultSharedPreferences(activityTestRule.getActivity());
+        for (String key : pm.getAll().keySet()) {
+            TelemetryWrapper.settingsClickEvent(key);
+        }
+    }
+
+    @Test
+    public void settingsLearnMoreClickEvent() {
+        activityTestRule.launchActivity(new Intent());
+        Context context = activityTestRule.getActivity();
+        TelemetryWrapper.settingsLearnMoreClickEvent(context.getString(R.string.pref_key_turbo_mode));
+        TelemetryWrapper.settingsLearnMoreClickEvent(context.getString(R.string.pref_key_telemetry));
```

Comment:
Instead of calling helper methods directly, I think it makes more sense to test the EventBuilder initialization in the helper methods.